### DISCUSSION TITLE
Draggable dice

### DISCUSSION THREAD

**Discussion by ClarinetK622** [seanmccarty/RequestedRolls#34] [Ideas]
Hey Sean,

Great work with this extension! I wouldn't run a table without it; members of my group don't want FGU's handy automation taking away their ability to make rolls themselves. While the extension is excellent, one challenge is that when the roll pops up, the only option is to click the button to throw virtual dice. A nice middle ground would be the option to click and drag dice into the chat box or dice tower (as a bonus it would enable hidden [SAVE VS] rolls). This would also be in service of what I believe this extension is designed to do, make for a more interactive experience at the virtual table.

As such, I've gone ahead and forked your repo and implemented draggable dice. I think this could be a nice feature to incorporate into the core extension for others to enjoy, but I will leave that up to you, as the mastermind of the original code. Feel free to take a look and see what you think. There may be bugs, or things done suboptimally; this was my first time working with Lua and xml, let alone cracking open an FGU extension. The way I've implemented things does add a level of complexity which may lead to unintended extension conflicts, so I understand if you'd rather keep it as a separate fork. In any case, I thought I would provide you the option. If you'd like, I'm happy to chat about squashing bugs and revising the code into something more suitable for incorporation into the RequestedRolls package.

https://github.com/ClarinetK622/RequestedRolls

- Justin

**Comment by seanmccarty** [comment]
I like the idea. The dice being there looks slick. It does run into a edge case when you feed it a dice string rather than a dice array so no dice show up. e.g. 2d6d1 (2d6 drop 1) I will have to think more about this because it breaks away from the standard procedure of how the manual dice window works. i.e. it assumes you already dragged the dice or clicked the button to roll.
There are a fair number of new overrides, so I will need to QC those for probable conflicts. It will be a while before I can get to this because of my day job.

**Reply by ClarinetK622** [reply]
Thanks for the quick reply! I actually discovered this edge case right after I published the idea and pushed a quick an dirty patch for the complex dice expressions (just throw in a d20 if no dice are populated). I’m sure there’s a way to do this right and populate at least the starting dice in a complex expression, but that should hold things over for now. Definitely will need QC. Keep me posted!

**Reply by ClarinetK622** [reply]
I should also note, if you cloned before I implemented the diecontrol patch you may have to clone again. I shuffled the branch name and rebased some things.

**Comment by seanmccarty** [comment]
@ClarinetK622 Can you please give my most recent commit a try? This is not on the Forge yet. I think the only thing that is missing is the fancy die control, but I think it otherwise does what you had in mind.

**Reply by ClarinetK622** [reply]
Sure thing, thanks! I’ve got a lot going on this week but I will find the time ASAP. Very much looking forward to it!

**Reply by seanmccarty** [reply]
I have done some more testing with the die control. The function is easy to add, but I think the clutter makes it overall worse. Plus, that control typically indicates you can add dice, and that is not my intention.
I plan to keep the existing interface, but the drag and drop will remain as features. I plan to release to the Forge after you have done some testing.

**Reply by ClarinetK622** [reply]
Sorry it took so long to check this out. Having played around with the most recent commit, everything seems to work as expected. Thank you for taking the time to implement the idea. I think it adds an extra level of polish to an already great extension. My players will love it. Props to you!

**Reply by ClarinetK622** [reply]
One thing I wonder. Is it possible to have the dice spawn when the button is clicked and held? This would bring it more in line with other click and drag roll interactions and make it more apparent that the dice can be released into the chat box to roll them. Just dragging the button icon doesn’t make that obvious.

**Reply by seanmccarty** [reply]
I pushed the commit for that. I will push it on the forge next I make a release.

**Reply by ClarinetK622** [reply]
Great stuff. Thanks!
